pytmc/xml_collector.py

```diff
@@ -956,308 +868,222 @@ def io_direction(self):
-                                else 'ArrayOut')
-
-        self.cfg.add_config_field("DTYP", '"{}"'.format(spec.data_type + direction_suffix))
-        return True
+        Generate the record to write values back to the PLC
 
-    @_skip_if_field_set('INP')
-    @_skip_if_field_set('OUT')
-    def guess_INP_OUT(self):
-        """
-        Construct, add, INP or OUT field
-        Fields will have this form:
-        "@asyn($(PORT),0,1)ADSPORT=851/Main.bEnableUpdateSine="
+        This will only be called if the ``io_direction`` is set to ``"output"`.
 
         Returns
         -------
-        bool
-            Return a boolean that is true iff a change has been made.
+        record: :class:`.EpicsRecord`
+            Description of output record
         """
+        # Base record with defaults
+        record = EPICSRecord(self.pvname,
+                             self.output_rtyp,
+                             fields=self.field_defaults)
+        # Add our port
+        record.fields['DTYP'] = self.dtyp
+        record.fields['OUT'] = self._asyn_port_spec + '="'
 
-        io, = self.cfg.get_config_lines('io')
-        io = io['tag']
-        name_list = self.chain.name_list
-        name = '.'.join(name_list)
-        assign_symbol = ""
-        field_type = ""
-        if 'i' in io and 'o' in io:
-            assign_symbol = "?"
-            if self.chain.last.is_array or self.chain.last.is_str:
-                field_type = 'INP'
-            else:
-                field_type = 'OUT'
-        elif 'i' in io:
-            assign_symbol = "?"
-            field_type = 'INP'
-        elif 'o' in io:
-            assign_symbol = "="
-            if self.chain.last.is_array or self.chain.last.is_str:
-                field_type = 'INP'
-            else:
-                field_type = 'OUT'
+        # Update with given pragamas
+        record.fields.update(self.cfg_as_dict().get('field', {}))
+        return record
 
-        try:
-            dtyp, = self.cfg.get_config_fields('DTYP')
-            dtyp = dtyp['tag']['f_set']
-        except (ValueError, KeyError):
-            return False
+    @property
+    def records(self):
+        """All records that will be created in the package"""
+        records = [self.generate_input_record()]
+        if self.io_direction == 'output':
+            records.append(self.generate_output_record())
+        return records
 
-        self.cfg.add_config_field(
-            field_type,
-            f'@asyn($(PORT),0,1)ADSPORT={self.ads_port}/{name}{assign_symbol}'
-        )
-        return True
 
-    @_skip_if_field_set('SCAN')
-    def guess_SCAN(self):
-        """
-        add field for SCAN field
+class BinaryRecordPackage(TwincatTypeRecordPackage):
+    """Create a set of records for a binary Twincat Variable"""
+    input_rtyp = 'bi'
+    output_rtyp = 'bo'
+    dtyp = '"asynInt32"'
+    field_defaults = {'ZNAM': '"Zero"', 'ONAM': '"One"'}
```

Comment:
```suggestion
    field_defaults = {'ZNAM': '"Zero"', 'ONAM': '"One"'}
```

@n-wbrown objections to changing this to "Off", "On"? I find a mapping of 0=Zero and 1=One to be not so helpful.

Reply:
Only minor ones. Given that this is just a default, I was thinking we should try to presume the least about our hypothetical user's use-case and try to leave things as literal as possible. I'm concerned we might be over interpreting intent especially given that this is just a bool. If you're strongly in favor of "On"/"Off" I'd say let's change it to "On"/"Off".